### PR TITLE
[doc] Terminology cleanup

### PR DESCRIPTION
**Pull request by msfschaffner**
This switches the "milestone" terminology used throughout test plannig to "(verification) stage". 

The change may warrant some more discussion in the Silicon WG, so please consider this as a proposal. However, I think it would be good to do this in order to disambiguate between the project milestones (M1/2 etc) and IP development / verification stages.

Also note that some re-vendoring back into and from the ibex repo needs to be done to get this fully aligned.

### CODE REVIEW

**Comment by matutem** [comment]
The regression report also shows a Milestone column, which should now be Stage. In the util/dvsim report generation.

File: util/dvsim/Testplan.py [hunk 1]
```diff
@@ -130,14 +130,14 @@ class Testpoint(Element):
     It captures following information:
     - name of the planned test
     - a brief description indicating intent, stimulus and checking procedure
-    - the targeted milestone
+    - the targeted stage
     - the list of actual developed tests that verify it
     """
     kind = "testpoint"
-    fields = Element.fields + ["milestone", "tests"]
+    fields = Element.fields + ["stage", "tests"]
```
Comment (msfschaffner) [inline util/dvsim/Testplan.py:137]: @matutem

**Comment by msfschaffner** [comment]
> The regression report also shows a Milestone column, which should now be Stage. In the util/dvsim report generation.

Indeed, this is supposed to fix that also (see the place where I have tagged you).

**Comment by matutem** [comment]
@msfschaffner There seem to be lots of references to milestone in util/dvsim: "grep -r -i milestone util/dvsim" ... You can create an issue if there are more than you can deal with now.

**Comment by msfschaffner** [comment]
> grep -r -i milestone util/dvsim

Thanks @matutem. I have replaced the instances that you pointed out, too.

**Comment by msfschaffner** [comment]
Note that the testplanner-related commits will have to be squashed together (the hjson and python-related changes have been separated for easier review).

**Comment by sriyerg** [comment]
Skimmed through - LGTM.

**Review by moidx — APPROVED** [review]

**Comment by msfschaffner** [comment]
Thanks, rebased!

**Pull request merged**